spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,60 @@
+require "rails_helper"
+
+describe SessionsController do
+  context "GET #new" do
+    context "without a password" do
+      it "does not set current user" do
+        user = create(:user)
+
+        post :create, params: { session: { email: user.email } }
+
+        expect(controller.current_user).to be nil
+      end
+
+      it "renders new" do
+        user = create(:user)
+
+        post :create, params: { session: { email: user.email } }
+
+        expect(response).to render_template(:new)
+      end
+
+      it "sets the flash[:alert]" do
+        user = create(:user)
+
+        post :create, params: { session: { email: user.email } }
+
+        expect(flash[:alert]).to match I18n.t("sessions.create.alert")
+      end
+    end
+
+    context "with a password" do
+      it "sets current user" do
+        user = create(:user)
+        allow(controller).to receive(:current_user).and_return(user)
+
```

Comment:
Actually this is my mistake... this should actually post to create with the correct params.
Because currently, it does not post anything, and we are stubbing the current user, and then check that out stub works.

Alternatively, if it is covered in a feature spec, this spec is not needed and can be removed.

Reply:
You're right, no it's not in a feature spec as I couldn't get current_user recognised there, so I will remove the stub - done in sha: 1fd80f8

Reply:
The feature spec to create a new session (spec/features/user_creates_new_session_spec.rb), and which checks that the user email is used checks this functionality. So suggesting to remove this spec :)

Reply:
Really, I thought I would need to literally use the words current_user for this to be the case!